go/vt/wrangler/tablet.go

```diff
@@ -88,7 +93,9 @@ func (wr *Wrangler) InitTablet(ctx context.Context, tablet *topodatapb.Tablet, a
 		if oldTablet.Keyspace != tablet.Keyspace || oldTablet.Shard != tablet.Shard {
 			return fmt.Errorf("old tablet has shard %v/%v. Cannot override with shard %v/%v. Delete and re-add tablet if you want to change the tablet's keyspace/shard", oldTablet.Keyspace, oldTablet.Shard, tablet.Keyspace, tablet.Shard)
 		}
-
+		if oldTablet.MasterTermStartTime != nil {
+			tablet.MasterTermStartTime = oldTablet.MasterTermStartTime
+		}
```

Comment:
This conflicts with setting it to `time.Now()` above, so I might need to re-think the logic.

Reply:
This is in fact correct. If we were already master, we read the old timestamp and use that, so that if a new master was elected during a master tablet restart, we don't override the new master.

Reply:
I thought we said that calling the low-level `vtctl InitTablet` to force a tablet to master implies an authoritative decision of who the master is. In that case, we should begin a new master term each time such a decision is communicated, just like with TER.

Reply:
oh yes, that's right. Restart only applies to the vttablet InitTablet.